astropy/units/tests/test_quantity.py

```diff
@@ -587,14 +587,9 @@ def test_comparison(self):
         assert not 1.0 * u.cm == 1.0
         assert 1.0 * u.cm != 1.0
 
-        # comparison with zero should raise a deprecation warning
         for quantity in (1.0 * u.cm, 1.0 * u.dimensionless_unscaled):
-            with pytest.warns(
-                AstropyDeprecationWarning,
-                match=(
-                    "The truth value of a Quantity is ambiguous. "
-                    "In the future this will raise a ValueError."
-                ),
+            with pytest.raises(
```

Comment:
Right now failing because of the change in error message -- I'd be fine with just matching on "ambiguous" and be done with it...